cmake/projects/x264/schemes/url_sha1_x264.cmake.in

```diff
@@ -0,0 +1,341 @@
+      -P
+      "@HUNTER_PACKAGE_SOURCE_DIR@/universal/${merge_lipo_script}"
+  )
+  set(ios_built_arch_roots)
+  set(multi_arch_install_root "@HUNTER_PACKAGE_BUILD_DIR@/multi-arch")
+  foreach(ios_architecture ${ios_architectures})
+    hunter_status_debug("Autotools: building for iOS architecture ${ios_architecture}")
+    #clear the conf options
+    set(is_simulator FALSE)
+    if(${ios_architecture} STREQUAL "armv7"
+        OR ${ios_architecture} STREQUAL "armv7s")
+      set(configure_host "arm-apple-darwin")
+    elseif(${ios_architecture} STREQUAL "arm64")
+      set(configure_host "aarch64-apple-darwin")
+    elseif(${ios_architecture} STREQUAL "i386")
+      set(configure_host "i386-apple-darwin")
+      set(is_simulator TRUE)
+    elseif(${ios_architecture} STREQUAL "x86_64")
+      set(configure_host "x86_64-apple-darwin")
+      set(is_simulator TRUE)
+    else()
+      hunter_fatal_error("iOS architecture: ${ios_architecture} not supported")
+    endif()
+
+    set(arch_flags)
+    # Extra space at the end of the arch_flags is needed below when appending
+    # to configure_opts, please do not remove!
+    if(is_simulator)
+      set(arch_flags "-arch ${ios_architecture} -isysroot ${IPHONESIMULATOR_SDK_ROOT} -miphoneos-version-min=${IOS_SDK_VERSION} ")
+    else()
+      set(arch_flags "-arch ${ios_architecture} -isysroot ${IPHONEOS_SDK_ROOT} -miphoneos-version-min=${IOS_SDK_VERSION} ")
+    endif()
+
+    # architecture specific source dir
+    set(arch_source_dir
+      "@HUNTER_PACKAGE_SOURCE_DIR@/multi-arch-build/${ios_architecture}"
+    )
+    set(arch_target
+        "${target_name}-${ios_architecture}"
+    )
+    set(arch_install_dir
+        "${multi_arch_install_root}/${ios_architecture}"
+    )
+
+    set(configure_command
+        ${shell_env}
+        ${x264_configure_command}
+        "--host=${configure_host}"
+        "--extra-cflags=${arch_flags}${cflags}"
+        "--extra-ldflags=${arch_flags}${ldflags}"
+        "--prefix=${arch_install_dir}"
+    )
+    ExternalProject_Add(${arch_target}
+        URL
+          @HUNTER_PACKAGE_URL@
+        URL_HASH
+          SHA1=@HUNTER_PACKAGE_SHA1@
+        DOWNLOAD_DIR
+          "@HUNTER_PACKAGE_DOWNLOAD_DIR@"
+        SOURCE_DIR
+          ${arch_source_dir}
+        INSTALL_DIR
+          ${arch_install_dir}
+          # not used, just avoid creating Install/<name> empty directory
+        CONFIGURE_COMMAND
+          ${configure_command}
+        BUILD_COMMAND
+          ${build_command}
+        BUILD_IN_SOURCE
+          1
+        INSTALL_COMMAND
+          ${install_command}
+    )
+
+    list(APPEND ios_built_arch_roots ${arch_install_dir})
+    add_dependencies(
+      "${ios_universal_target}"
+      "${arch_target}"
+    )
+  endforeach()
+
+  set(HUNTER_PACKAGE_INSTALL_PREFIX "@HUNTER_PACKAGE_INSTALL_PREFIX@")
+  configure_file(
+      "@HUNTER_GLOBAL_SCRIPT_DIR@/${merge_lipo_script}.in"
+      "@HUNTER_PACKAGE_SOURCE_DIR@/universal/${merge_lipo_script}"
```

Comment:
Ah, ok, I think I understand why it works. `@HUNTER_PACKAGE_SOURCE_DIR@/universal` is used as a source directory of a `ExternalProject_Add(...)` call without a file to be extracted, and it is before extracting the file that the directory is removed. I guess in case one wants to use a path as a source directory, which seems to be legal according to the documentation: https://cmake.org/cmake/help/v3.0/module/ExternalProject.html .

`If SOURCE_DIR is explicitly set to an existing directory the project will be built from it(...)` 

I skimmed through the source of `ExternalProject.cmake` and it seems that it is before extracting the file of an EP that the directory is removed.

In the iOS build, it happens that we use `@HUNTER_PACKAGE_SOURCE_DIR@/multi-arch-build/${ios_architecture}` for each iOS architecture as the source of the EP calls with the package files to be extracted, so the `@HUNTER_PACKAGE_SOURCE_DIR@/universal` is never touched. So I think we are safe.

Since you had this doubt in while reviewing it shows that this code is rather non-trivial, I actually tried to extract this behavior into a function to be reused in other packages when integrating other build systems, but found it rather hard. First need to treat the running commands with specific env vars, then I guess this can be looked upon.

> Let's test it. With what package does it work?

`odb-sqlite` and its dependencies `SQLite3` and `odb` used to work